ic-os/components/ic/generate-replica-config.sh

```diff
@@ -58,116 +103,39 @@ function get_if_address_retries() {
+    [ "${QUERY_STATS_EPOCH_LENGTH}" = "null" ] && QUERY_STATS_EPOCH_LENGTH="600"  # Default is 600 blocks (around 10min)
+    [ "${JAEGER_ADDR}" = "null" ] && JAEGER_ADDR=""
 
-# Read the backup config variables from file. The file must be of the form
-# "key=value" for each line with a specific set of keys permissible (see
-# code below).
-#
-# Arguments:
-# - $1: Name of the file to be read.
-function read_backup_variables() {
-    # Read limited set of keys. Be extra-careful quoting values as it could
-    # otherwise lead to executing arbitrary shell code!
-    while IFS="=" read -r key value; do
-        case "$key" in
-            "backup_retention_time_secs") backup_retention_time_secs="${value}" ;;
-            "backup_puging_interval_secs") backup_purging_interval_secs="${value}" ;;
-        esac
-    done <"$1"
+    # todo: remove node_index variable and hard-code into ic.json5.template
+    NODE_INDEX="0"
 }
 
-# Read malicious behavior config variables from file. The file must be of the
-# form "key=value" for each line with a specific set of keys permissible (see
-# code below).
-#
-# Arguments:
-# - $1: Name of the file to be read.
-function read_malicious_behavior_variables() {
-    # Read limited set of keys. Be extra-careful quoting values as it could
-    # otherwise lead to executing arbitrary shell code!
-    while IFS="=" read -r key value; do
-        case "$key" in
-            "malicious_behavior") malicious_behavior="${value}" ;;
-        esac
-    done <"$1"
-}
-
-# Read query stats config variables from file. The file contains a single value which is the epoch length in seconds.
-function read_query_stats_variables() {
-    while IFS="=" read -r key value; do
-        case "$key" in
-            "query_stats_epoch_length")
-                query_stats_epoch_length="${value}"
-                ;;
-        esac
-    done <"$1"
-}
-
-# Read Jaeger address variable from file. The file contains a single value Jaeger node address used in system tests.
-function read_jaeger_addr_variable() {
-    while IFS="=" read -r key value; do
-        case "$key" in
-            "jaeger_addr") jaeger_addr="${value}" ;;
-        esac
-    done <"$1"
+# If the URL is of the form "https://[IPv6]/" then we extract the IPv6 address.
+# Otherwise, we copy over the whole URL to ic.json5
+# If a URL is of the form "http://[IPv6_address]/" or "https://[IPv6_address]/", 
+# it extracts just the IPv6 address.
+# For all other URLs, it leaves them unchanged.
+function process_nns_urls() {
+    local processed_urls=()
+    IFS=',' read -ra URLS <<< "$NNS_URLS"
+    for url in "${URLS[@]}"; do
+        if [[ $url =~ ^https?://\[([0-9a-fA-F:]+)\](:[0-9]+)?(/.*)?$ ]]; then
+            ipv6_addr="${BASH_REMATCH[1]}"
+            processed_urls+=("$ipv6_addr")
+        else
+            # Keep the URL as is
+            processed_urls+=("$url")
+        fi
+    done
+    NNS_URLS=$(IFS=','; echo "${processed_urls[*]}")
 }
```

Comment:
In some testing, we pass ipv6 addresses directly as the values for nns_url (I'm not entirely certain why).

Now that we are going through the config_tool, which uses URL types to hold the nns_urls, we must wrap those ipv6 addresses like:
format!("http://[{}]", ipv6_addr)
(see launch-single-vm/src/main.rs)

We need to then unwrap the ipv6 values before substituting the values in ic.json5 though. For "normal" nns_urls, like the ones used on production, we don't need to modify them